mat64/dense.go

```diff
@@ -113,6 +113,31 @@ func (m *Dense) reuseAs(r, c int) {
 	}
 }
 
+// untranspose untransposes a matrix if applicable. If a is an Untransposer, then
+// untranspose returns the underlying matrix and true. If it is not, then it returns
+// the input matrix and false.
+func untranspose(a Matrix) (Matrix, bool) {
+	if ut, ok := a.(Untransposer); ok {
+		return ut.Untranspose(), true
+	}
+	return a, false
+}
+
+// uniqueMemoryDense returns m with different memory from a. It also returns
+// a callback to defer for cleanup at the return of the call if necessary.
+func uniqueMemoryDense(m *Dense, a Matrix) (*Dense, func()) {
```

Comment:
I like making it a method and using the word workspace, but I don’t like workspaceWith. The “With” doesn’t imply anything. Maybe “isolateWorkspace” or something like that?

> On Jul 27, 2015, at 6:20 PM, Dan Kortschak <notifications@github.com> wrote:
> 
> In mat64/dense.go <https://github.com/gonum/matrix/pull/157#discussion_r35603277>:
> 
> > @@ -113,6 +113,31 @@ func (m *Dense) reuseAs(r, c int) {
> >  	}
> >  }
> >  
> > +// untranspose untransposes a matrix if applicable. If a is an Untransposer, then
> > +// untranspose returns the underlying matrix and true. If it is not, then it returns
> > +// the input matrix and false.
> > +func untranspose(a Matrix) (Matrix, bool) {
> > +	if ut, ok := a.(Untransposer); ok {
> > +		return ut.Untranspose(), true
> > +	}
> > +	return a, false
> > +}
> > +
> > +// uniqueMemoryDense returns m with different memory from a. It also returns
> > +// a callback to defer for cleanup at the return of the call if necessary.
> > +func uniqueMemoryDense(m *Dense, a Matrix) (*Dense, func()) {
> Make it a method on *Dense and call it workspaceWith.
> —
> Reply to this email directly or view it on GitHub <https://github.com/gonum/matrix/pull/157/files#r35603277>.
>